libcperciva/network/network.h

```diff
@@ -41,12 +41,13 @@ void * network_connect(struct sock_addr * const *,
 
 /**
  * network_connect_bind(sas, sa_b, callback, cookie):
- * Iterate through the addresses in ${sas}, attempting to create and connect
- * a non-blocking socket after binding to ${sa_b}.  Once connected, invoke
- * ${callback}(${cookie}, s) where s is the connected socket; upon fatal error
- * or if there are no addresses remaining to attempt, invoke
- * ${callback}(${cookie}, -1).  Return a cookie which can be passed to
- * network_connect_cancel() in order to cancel the connection attempt.
+ * Iterate through the addresses in ${sas}, attempting to create and connect a
+ * non-blocking socket.  If ${sa_b} is not NULL, then bind the socket to
```

Comment:
The binding happens before connecting, so this isn't right.

Reply:
Hmm.  It would be nice if the description of `network_connect_bind` remained fairly similar to that of `network_connect`.

How about "If sa_b is not NULL, then bind the socket to sa_b immediately after creating it"?

I've also modified the API comment for `sock_connect_bind_nb`, which has the same problem.

The alternative (if we don't add something like "immediately after creating it") for `sock_connect_bind_nb` would be:

> Create a socket.  If ${sa_b} is not NULL, then bind the socket to ${sa_b}.  Mark the socket as non-blocking, and attempt to connect to the address ${sa}.

I mean, sure that's a literal description of what's happening in the `.c` file, but it's rather awkward.